File: Easy/Padovan Sequence.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution
{
    public:
    int padovanSequence(int n)
    {
       //code here
       if (n < 3) return 1;
       long long int ans = 0;
       
       long long int m = pow(10, 9) +7;
       
       long long int p0 = 1;
       long long int p1 = 1;
       long long int p2 = 1;
       
       for (int i=2; i<n; i++)
       {
            ans = (p0 %m + p1 %m) %m;
            p0 = p1;
            p1 = p2;
            p2 = ans;
       }
       
       return (int)ans %m;
    }
    
};


//{ Driver Code Starts.
int main()
{
    int t;
    cin>>t;
    while(t--)
    {
    	int n;
    	cin>>n;
    	Solution ob;
    	cout<<ob.padovanSequence(n)<<endl;
    }
    return 0;
}
// } Driver Code Ends

File: Easy/Padovan Sequence.cs
//{ Driver Code Starts
// Initial Template for C#

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DriverCode
{

    class GFG
    {
        static void Main(string[] args)
        {
            int testcases;// Taking testcase as input
            testcases = Convert.ToInt32(Console.ReadLine());
            while (testcases-- > 0)// Looping through all testcases
            {
                int N = Convert.ToInt32(Console.ReadLine());// Input for size of array
                Solution obj = new Solution();
                int res = obj.padovanSequence(N);
                Console.Write(res);
                Console.Write("\n");
          }

        }
    }
}
// } Driver Code Ends


//User function Template for C#
class Solution
    {
        //Complete this function
        public int padovanSequence(int n)
        {
            //code here
            int m = 1000000007;
            
            if (n <= 2)
            {
                return 1;
            }
           
            int a = 1;
            int b = 1;
            int c = 1;
           
            int ans = 2;
           
            for (int i = 3; i <= n; i++)
            {
                ans = (a + b) % m;
                
                a = b;
                b = c;
                c = ans;
            }
            
            return ans;
           
           
           
        }
        
    }